src/Umbraco.Core/Services/IdkMap.cs

```diff
@@ -139,10 +144,12 @@ public void ClearCache(int id)
             try
             {
                 _locker.EnterWriteLock();
-                Guid key;
-                if (_id2Key.TryGetValue(id, out key) == false) return;
-                _id2Key.Remove(id);
-                _key2Id.Remove(key);
+                var match = _id2Key.Keys.SingleOrDefault(x => x.Id == id);
```

Comment:
Simply because I couldn't make it return anything to begin with - which turned out to be because I didn't implement the comparison members ... I'll change it to objects now instead :)